src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -228,9 +235,21 @@ private static int GetMaxTaskCount(RestoreArgs restoreContext)
 
         private static async Task<RestoreSummary> ExecuteAndCommitAsync(RestoreSummaryRequest summaryRequest, CancellationToken token)
         {
-            var result = await ExecuteAsync(summaryRequest, token);
+            using (var telemetry = TelemetryActivity.Create(parentId: summaryRequest.Request.ParentId, eventName: ProjectExecuteAndCommitAsyncInformation))
```

Comment:
I'm not excited about adding yet another event in this codepath. 
https://github.com/NuGet/Client.Engineering/issues/319 calls for consolidating the restore telemetry into one event. I don't necessarily want to block this PR ofc, because it's easy to refactor around it, there's no customer contract, but something to keep in mind.

Reply:
The solution might be starting the event *here* and then just the restore command adds measurements.